crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -180,4 +194,265 @@ impl<'a> Compressor<'a> {
+                Some(1.0)
+            } else {
+                Some(0.0)
+            }
+        }
+        Expression::NullLiteral(_) => Some(0.0),
+        _ => None,
+    }
+}
+
+/// code port from [closure compiler](https://github.com/google/closure-compiler/blob/a4c880032fba961f7a6c06ef99daa3641810bfdd/src/com/google/javascript/jscomp/base/JSCompDoubles.java#L113)
+/// <https://262.ecma-international.org/5.1/#sec-9.5>
+#[allow(clippy::cast_possible_truncation)] // for `as i32`
+fn ecmascript_to_int32(num: f64) -> i32 {
+    let int32_value = num as i32;
+    if (f64::from(int32_value) - num).abs() < f64::EPSILON {
+        return int32_value;
+    }
+
+    let pos_int = num.signum() * num.abs().floor();
+    let int32bit = pos_int % 2f64.powi(32);
+
+    if int32bit >= 2f64.powi(31) { (int32bit - 2f64.powi(32)) as i32 } else { int32bit as i32 }
+}
+
+fn get_boolean_value(expr: &Expression) -> Tri {
+    let be_tri_boolean = |boolean: bool| {
+        if boolean { Tri::True } else { Tri::False }
+    };
+
+    match expr {
+        Expression::RegExpLiteral(_) | Expression::ArrayExpression(_)| Expression::ArrowExpression(_)| Expression::ClassExpression(_) | Expression::FunctionExpression(_)| Expression::NewExpression(_) |  Expression::ObjectExpression(_) => Tri::True,
+        Expression::NullLiteral(_) => Tri::False,
+        Expression::BooleanLiteral(boolean_literal) => {
+            be_tri_boolean(boolean_literal.value)
+        },
+        Expression::NumberLiteral(number_literal) => {
+            be_tri_boolean(number_literal.value != 0.0)
+        },
+        Expression::BigintLiteral(big_int_literal) => be_tri_boolean(big_int_literal.value == BigUint::default()),
+        Expression::StringLiteral(string_literal) => be_tri_boolean(!string_literal.value.is_empty()),
+        Expression::TemplateLiteral(template_literal) => {
+            if let Some(quasi) = template_literal.quasis.get(0) && quasi.tail {
+                if quasi.value.cooked.as_ref().map_or(false, |cooked| !cooked.is_empty()) {
+                    Tri::True
+                } else {
+                    Tri::False
+                }
+            } else {
+                Tri::Unknown
+            }
+        },
+        Expression::Identifier(ident) => {
+            if expr.is_undefined() {
+                Tri::False
+            } else if  ident.name == "Infinity" {
+                Tri::True
+            } else if ident.name == "NaN" {
+                Tri::False
+            } else {
+                Tri::Unknown
+            }
+        },
+        Expression::AssignmentExpression(assign_expr) => {
+            match assign_expr.operator {
+                // TODO: Is there possible to be true or false ?
+                // https://github.com/google/closure-compiler/blob/a4c880032fba961f7a6c06ef99daa3641810bfdd/src/com/google/javascript/jscomp/NodeUtil.java#L186 
+                AssignmentOperator::LogicalAnd | AssignmentOperator::LogicalOr => {
```

Comment:
Is there a possible to be `true` or `false` ?

https://github.com/google/closure-compiler/blob/a4c880032fba961f7a6c06ef99daa3641810bfdd/src/com/google/javascript/jscomp/NodeUtil.java#L186